certbot/interfaces.py

```diff
@@ -238,6 +238,12 @@ class IConfig(zope.interface.Interface):
         "This only affects the port Certbot listens on. "
         "A conforming ACME server will still attempt to connect on port 80.")
 
+    http01_use_tls = zope.interface.Attribute(
+        "Runs the standalone server with a self signed certificate. "
+        "Useful for when you need to validate an http-01 challenge with "
+        "an https endpoint or if you have an upstream proxy redirecting "
+        "all http traffic to https.")
```

Comment:
You are right on point and we are talking about the same thing. I'll start from where I differ a little bit, although I think your solution might be the right one, thank you.

Step 5: Reverse proxy (Haproxy) uses TLS-SNI based routing to route requests based on server name indication to the correct upstream/backend server. The upstream/backend servers are running nginx, but they also have the ACME client installed. I have a bootstrap script that calls the ACME client before it attempts to start nginx. I use the standalone option with http-01 to try and obtain these certificates and once I have them, I start nginx with the obtained certificates. 

I think you have solved my problem at @patf, I need to first start nginx running as an https reverse proxy so that when the https request arrives on the server, I simply proxy it to the certbot client running locally on port 80 serving the http-01 challenge. 

That was a good exploration into the `certbot` code base and I'm glad there is a solution to what I'm trying to do. Thanks again.